src/aux-runtime/runtime/AuxLibrary.spec.ts

```diff
@@ -9985,11 +9984,113 @@ describe('AuxLibrary', () => {
+                };
+
                 expect(promise[ORIGINAL_OBJECT]).toEqual(expected);
                 expect(context.actions).toEqual([expected]);
             });
+
+            it('should increment task ID for multiple calls', () => {
+                const initialTaskSize = context.tasks.size;
+
+                const promise1: any = library.api.os.removeMapLayer(
+                    bot1,
+                    'layer1'
+                );
+                const expected1 = {
+                    type: 'remove_bot_map_layer',
+                    botId: 'test1',
+                    overlayId: 'layer1',
+                    taskId: initialTaskSize + 1,
+                };
+
+                const promise2: any = library.api.os.removeMapLayer(
+                    bot1,
+                    'layer2'
+                );
+                const expected2 = {
+                    type: 'remove_bot_map_layer',
+                    botId: 'test1',
+                    overlayId: 'layer2',
+                    taskId: initialTaskSize + 2,
+                };
+
+                expect(promise1[ORIGINAL_OBJECT]).toEqual(expected1);
+                expect(promise2[ORIGINAL_OBJECT]).toEqual(expected2);
+                expect(context.actions).toEqual([expected1, expected2]);
+            });
+        });
+
+        describe('os.removeMapLayer() - Extended Tests', () => {
+            beforeEach(() => {
+                context.actions = [];
+            });
+
+            it('should work with map portal', () => {
+                const promise: any = library.api.os.removeMapLayer(
+                    bot1,
+                    'map-layer'
+                );
+
+                const expected = {
+                    type: 'remove_bot_map_layer',
+                    botId: 'test1',
+                    overlayId: 'map-layer',
+                    taskId: context.tasks.size,
+                };
+
+                expect(promise[ORIGINAL_OBJECT]).toEqual(expected);
+                expect(context.actions).toContainEqual(expected);
+            });
+
+            it('should work with miniMap portal', () => {
+                const promise: any = library.api.os.removeMapLayer(
+                    bot1,
+                    'minimap-layer'
+                );
+                const expected = {
+                    type: 'remove_bot_map_layer',
+                    botId: 'test1',
+                    overlayId: 'minimap-layer',
+                    taskId: context.tasks.size,
+                };
+
+                expect(promise[ORIGINAL_OBJECT]).toEqual(expected);
+                expect(context.actions).toContainEqual(expected);
+            });
+
+            it('should return a promise', () => {
+                const result = library.api.os.removeMapLayer(
+                    bot1,
+                    'test-layer'
+                );
+                expect(result).toBeInstanceOf(Promise);
+            });
         });
```

Comment:
@TroyceGowdy these tests (`addMapLayer()` and `removeMapLayer()`) should not be changed by this PR. Instead, please make sure that these tests are unaffected by your pull request, and that there are additional tests for the functions you are adding: `addBotMapLayer()` and `removeBotMapLayer()`.